llvm/utils/lit/tests/shtest-ulimit.py

```diff
@@ -3,13 +3,8 @@
 # ulimit does not work on non-POSIX platforms.
 # UNSUPPORTED: system-windows
 
-# TODO(boomanaiden154): The test fails on some non-Linux POSIX
-# platforms (like MacOS) due to the underlying system not supporting
-# ulimit -v. This test needs to be carved up so we keep full test
-# coverage on Linux and as much as possible on other platforms.
-# REQUIRES: system-linux
-
 # RUN: not %{lit} -a -v %{inputs}/shtest-ulimit | FileCheck %s
+# RUN: %if system-linux %{ not %{lit} -a -v %{inputs}/shtest-ulimit-linux | FileCheck %s --check-prefix CHECK-LINUX %}
```

Comment:
This will also work on other operating systems such as FreeBSD. Could we split this test into two and just make the one that has `ulimit -v` `UNSUPPORTED: system-mac`?

Reply:
Done.